octavia-cli/build.gradle

```diff
@@ -7,3 +7,12 @@ airbytePython {
     moduleDirectory 'octavia_cli'
 }
 
+task generateApiClient(type: Exec) {
```

Comment:
does the caching work properly on this task? the expectation should be that if `airbyte-api/src/main/openapi/config.yaml` does not change that this task does not need to run again. if that's not set up properly it will require re-executing this task and every tasks that depends on them (which is most of them).

see `UP-TO-DATE` in https://docs.gradle.org/current/userguide/more_about_tasks.html

Reply:
Thanks for this suggestion! I realized I could use a grade plugin already used for java API client generation: `org.openapitools.generator.gradle.plugin.tasks.GenerateTask`. It supports incremental builds, but I can't figure why it's not working in my context.
I did the following:
* update `octavia-cli/build.gradle` to use the GenerateTask.
* exclude the generated client from the license headers checks.
* remove the script I wrote to generate the client with the docker image.

Still, I don't get why I can't see the `UP-TO-DATE` flag after consequent builds.

Reply:
I finally succeeded in making the `FROM-CACHE` and `UP-TO-DATE` mechanisms work. The solution was to use the `buildDir` as output dir, using other paths made gradle re-run the task on each run. The generated client is now stored in `octavia-cli/build/airbyte_api_client`. Related commit: https://github.com/airbytehq/airbyte/commit/e95abc9ccc853b84ed8cfd5f35223b45577566ab

Reply:
awesome!

Reply:
I'd like you to present this very briefly in the dev meeting. This is something we often get wrong when using gradle, so it would be good to have a 5 minute presentation to share the knowledge with the team.

Reply:
ok! To be honest, I'm not understanding why using input paths for tasks that are not `buildDir` does not lead to `UP-TO-DATE` but it could indeed be interesting to share this finding and to open this discussion in the dev meeting.